### ISSUE TITLE
Point dc.load returns Null

### ISSUE THREAD

**Issue by simonaoliver** [opendatacube/datacube-core#341]
### Expected behaviour
A point query on a indexed GeoTIFF should return similar values to it's ingested product

### Actual behaviour
Point query results returns -999 on GeoTIFF point query

### Steps to reproduce the behaviour
Ingested data point query returns as expected:

```pycon
    >>> query = {
    ...    'lat': (-20.2),
    ...    'lon': (145.8),
    ... }
    >>> ls5nbar = dc.load(product='ls5_nbar_albers', **query)
    >>> ls7nbar.red.mean()
    >>> <xarray.DataArray 'red' ()>
    array(1310.9008746355685)
```

Indexed data in GeoTIFF format returns -999 (product above is an ingested version of the product demonstrated below):

```pycon
    >>> query = {
    ...    'lat': (-20.2),
    ...    'lon': (145.8),
    ...    'output_crs': 'EPSG:4326',
    ...    'resolution': (-0.0006,0.0006)
    ... }
    >>> ls5nbarscene = dc.load(product='ls5_nbar_scene', **query)
    >>> ls5nbarscene['3'].mean()
    <xarray.DataArray '3' ()>
    array(-999.0)
```
Expect to see the same / similar mean value here i.e. not -999

    <xarray.Dataset>
    Dimensions:    (latitude: 1, longitude: 1, time: 344)
    Coordinates:
      * time       (time) datetime64[ns] 1986-08-17T23:38:44.500000 ...
      * latitude   (latitude) float64 -20.2
      * longitude  (longitude) float64 145.8
    Data variables:
        1          (time, latitude, longitude) int16 -999 -999 -999 -999 -999 ...
        2          (time, latitude, longitude) int16 -999 -999 -999 -999 -999 ...
        3          (time, latitude, longitude) int16 -999 -999 -999 -999 -999 ...
        4          (time, latitude, longitude) int16 -999 -999 -999 -999 -999 ...
        5          (time, latitude, longitude) int16 -999 -999 -999 -999 -999 ...
        7          (time, latitude, longitude) int16 -999 -999 -999 -999 -999 ...
    Attributes:
        crs:      EPSG:4326

### Environment information
NCI VDI using `datacube.__version__ = '1.5.4'`

**Comment by omad** [comment]
Thanks Simon. Sounds pretty serious for any _pixel drill_ applications. 

@vnewey has discovered the same issue, I'm not sure if she has managed to dig into or start fixing it.

**Comment by Kirill888** [comment]
Problem is not specific to GeoTiff, rather point load is broken whenever output crs differs from native. It's just that with scene products you must specify CRS since there is no global product CRS and so you are more likely to hit this problem.

Load should probably detect point queries and rather than re-projecting a `1x1` raster into a different coordinate frame, instead should transform query point to the native projection of the source file and read that coordinate without re-projection.

**Comment by clausmichele** [comment]
Are there any progress on this?

**Comment by clausmichele** [comment]
Is there any progress on this?

**Comment by Kirill888** [comment]
@clausmichele not really, may I ask how it impacts you? I did do PixelDrill class at some point, but never got time to properly make it usable remains of that are here:
https://github.com/opendatacube/odc-tools/blob/master/old/dea/io/pdrill.py
